waywiser_hwbringup/include/point_cloud_transformer.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef WAYWISER_HWBRINGUP_POINT_CLOUD_TRANSFORMER_HPP_
+#define WAYWISER_HWBRINGUP_POINT_CLOUD_TRANSFORMER_HPP_
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/point_cloud2.hpp"
+#include "tf2_ros/buffer.h"
+#include "tf2_ros/transform_listener.h"
+#include "tf2_sensor_msgs/tf2_sensor_msgs.hpp"
+#include "tf2/exceptions.h"
+#include "geometry_msgs/msg/transform_stamped.hpp"
+
+namespace waywiser_hwbringup
+{
+class PointCloudTransformer : public rclcpp::Node
```

Comment:
I would like to have a short comment about what it does and why it is needed somewhere here in the actual node, not only launch file. Is this an actual "fix" and realsense specific or does it need to be applied to any depth camera (looks like the latter but not entirely sure)?
Are there any benefits distributing this node to three files instead of one (max two) considering its small size?

Reply:
It is actually a fix for correct visualization of point clouds in rviz2. To represent the points using their depth value for their color intensity, we would have to choose "z" axis instead of "x" axis (as should be done if the data is in its ros2 base frame). This seems to be the case with the realsense depth camera where the reference frame id of point cloud is not its base frame ("camera_link"). I am not sure if it is the case with other depth cameras. 

This fix is not otherwise required since this conversion is done where the point cloud is used. For example, when using collision monitor, it is done here: https://github.com/ros-planning/navigation2/blob/b489f5b79baea6c9f6b91ecb23c5a2725317c140/nav2_collision_monitor/src/pointcloud.cpp#L112